### DISCUSSION TITLE
Adding the gun bag functionality to the launcher slot

### DISCUSSION THREAD

**Discussion by Tirpitz93** [acemod/ACE3#8043] [Ideas]
Hello,
I had a look at the Gun bag for the first time today. Really like the feature, but I feel like the gun bag functionality would be better or additionally located as a launcher slot (like CSW weapon bags).

Really like the self interact to take/exchange the weapon. 
This would allow a sniper team to carry a light weapon into the AO, and when in situ bring out the big gun, as well as carrying a backpack. This would not be much different from having a launcher on your back from an ergonomics point of view.

**Comment by BrettMayson** [comment]
This will not be possible due to a technical limitation of the engine.

Backpacks are their own separate objects, which means we can store information on them using [setVariable](https://community.bistudio.com/wiki/setVariable). ACE stores the gun type using this command, which is not possible on a launcher as they do not have this functionality. There would be no way for ACE to know which gun is supposed to be stored inside the bag.

**Reply by Tirpitz93** [reply]
I mean it's always a matter of how complicated are you willing to make a system in order for it to work. you could attach the variables to the player, and on dropping it, transfer them to the ground holder, then to the next player.

**Reply by Tirpitz93** [reply]
Another option would be to handle it like TFAR,  with _IDd classes up to 1000 (or whatever).

**Reply by BrettMayson** [reply]
In my opinion it's not worth the hassle of either of those options. I rarely see the gun bags used to begin with, and using IDs can be very inconvenient. With persistent systems they are extremely annoying, TFAR / ACRE are worth the hassle for the amount they are used and how much they improve the gameplay experience, I don't think gun bags are worth it. If you want to be able to carry more gear while using a gun bag you could always increase the capacity in a patch mod.
